cluster/addons/fluentd-gcp/fluentd-gcp-image/google-fluentd.conf

```diff
@@ -16,72 +21,97 @@
-#   kubernetes.synthetic-logger-0.25lps-pod_default-synth-lgr
+#
+#  var.log.containers.synthetic-logger-0.25lps-pod_default_synth-lgr-997599971ee6366d4a5920d25b79286ad45ff37a74494f262e3bc98d909d0a7b.log
+#
+# The Kubernetes fluentd plugin is used to extract the namespace, pod name & container name
+# which are added to the log message as a kubernetes field object & the Docker container ID
+# is also added under the docker field object.
+# The final tag is:
+#
+#   kubernetes.var.log.containers.synthetic-logger-0.25lps-pod_default_synth-lgr-997599971ee6366d4a5920d25b79286ad45ff37a74494f262e3bc98d909d0a7b.log
+#
+# And the final log record look like:
+#
+# {
+#   "log":"2014/09/25 21:15:03 Got request with path wombat\n",
+#   "stream":"stderr",
+#   "time":"2014-09-25T21:15:03.499185026Z",
+#   "kubernetes": {
+#     "namespace": "default",
+#     "pod_name": "synthetic-logger-0.25lps-pod",
+#     container_name: "synth-lgr"
+#   },
+#   "docker": {
+#     "container_id": "997599971ee6366d4a5920d25b79286ad45ff37a74494f262e3bc98d909d0a7b"
+#   }
+# }
+#
 # This makes it easier for users to search for logs by pod name or by
 # the name of the Kubernetes container regardless of how many times the
 # Kubernetes pod has been restarted (resulting in a several Docker container IDs).
+#
 # TODO: Propagate the labels associated with a container along with its logs
 # so users can query logs using labels as well as or instead of the pod name
-# and container name.
+# and container name. This is simply done via configuration of the Kubernetes
+# fluentd plugin but requires secrets to be enabled in the fluent pod. This is a
+# problem yet to be solved as secrets are not usable in static pods which the fluentd
+# pod must be until a per-node controller is available in Kubernetes.
+# This configuration file for Fluentd / td-agent is used
+# to watch changes to Docker log files that live in the
+# directory /var/log/docker/containers/ and are symbolically
+# linked to from the /var/log/containers directory using names that capture the
+# pod name and container name. These logs are then submitted to
 
 <source>
   type tail
-  format none
+  format json
   time_key time
-  path /varlog/containers/*.log
-  pos_file /varlog/gcp-containers.log.pos
+  path /var/log/containers/*.log
+  pos_file /var/log/gcp-containers.log.pos
   time_format %Y-%m-%dT%H:%M:%S
-  tag reform.*
+  tag kubernetes.*
   read_from_head true
 </source>
 
-<match reform.**>
-  type record_reformer
-  enable_ruby true
-  tag kubernetes.${tag_suffix[3].split('-')[0..-2].join('-')}
-</match>
-
-<match kubernetes.**>
-  type google_cloud
-  flush_interval 5s
-  # Never wait longer than 5 minutes between retries.
-  max_retry_wait 300
-  # Disable the limit on the number of retries (retry forever).
-  disable_retry_limit
-</match>
-
 <source>
   type tail
   format none
   time_key time
-  path /varlog/kubelet.log
-  pos_file /varlog/gcp-kubelet.log.pos
+  path /var/log/kubelet.log
```

Comment:
This was just copied from the previous log, just mounting /var/log on host to same location in container. I don't think kubelet logs to here, probably to journald I think, but didn't want to remove it in case it was something you knew about.